packages/spatial/src/renderer/components/HighlightComponent.ts

```diff
@@ -26,23 +26,32 @@ Ethereal Engine. All Rights Reserved.
 import { defineQuery, defineSystem, Engine } from '@etherealengine/ecs'
 import { defineComponent, getComponent, hasComponent } from '@etherealengine/ecs/src/ComponentFunctions'
 
+import { Object3D } from 'three'
+import { traverseEntityNode } from '../../transform/components/EntityTree'
 import { RendererComponent, WebGLRendererSystem } from '../WebGLRendererSystem'
 import { GroupComponent } from './GroupComponent'
 import { MeshComponent } from './MeshComponent'
 import { VisibleComponent } from './VisibleComponent'
 
 export const HighlightComponent = defineComponent({ name: 'HighlightComponent' })
 
-const highlightQuery = defineQuery([HighlightComponent, MeshComponent, GroupComponent, VisibleComponent])
+const highlightQuery = defineQuery([HighlightComponent, VisibleComponent])
 
 const execute = () => {
   /** @todo support multiple scenes */
   if (!hasComponent(Engine.instance.viewerEntity, RendererComponent)) return
 
+  let highlightObjects: Object3D[] = []
```

Comment:
If you change this type to a Set<Object3D> it'll remove the need for the ts-ignore below